### PR TITLE
Test filtering

### PR DESCRIPTION
**Pull request by nijotz**
## Proposed Release Notes

* More precise filtering of leveraging secrets.

### CODE REVIEW

File: bin/ssl.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ set -x  # be chatty and show the lines we're running
 # that are mysterious and not understood, so let
 # bail early if we detect that's the case
 ENGINE_OPENSSL=`openssl version | awk '{print $(1)}'`
-if [ "$ENGINE_OPENSSL" == "LibreSSL" ]
+if [ "$ENGINE_OPENSSL" = "LibreSSL" ]
```
Comment (nijotz) [inline bin/ssl.sh:11]: Noticed some shell errors on github actions when running on ubuntu. Saw that the `==` syntax isn't cross-shell compatible. Same with `-a` vs `-e`

Comment (nijotz) [reply bin/ssl.sh:11]: These changes fixed those errors

File: package.json [hunk 2]
```diff
@@ -163,7 +162,7 @@
   },
   "devDependencies": {
     "@newrelic/proxy": "^2.0.0",
-    "@newrelic/test-utilities": "^3.1.0",
+    "@newrelic/test-utilities": "^3.3.0",
```
Comment (nijotz) [inline package.json:165]: The new test utilities supports skipping using multiple keywords

File: bin/run-versioned-tests.sh [hunk 3]
```diff
@@ -21,5 +21,14 @@ fi
...
+
+# Don't run the aws-sdk tests if we don't have the keys set
+if [[ -z "$AWS_ACCESS_KEY_ID" ]]; then
+  skip="${skip},aws-sdk"
+fi
+
+
 # This is meant to be temporary. Remove once new major version with fixes rolled into agent.
-time ./node_modules/.bin/versioned-tests $VERSIONED_MODE -i 2 -s koa ${directories[@]}
+time ./node_modules/.bin/versioned-tests $VERSIONED_MODE -i 2 -s $skip ${directories[@]}
```
Comment (nijotz) [inline bin/run-versioned-tests.sh:34]: We're able to skip with multiple keywords now

File: bin/run-versioned-tests.sh [hunk 4]
```diff
@@ -21,5 +21,14 @@ fi
 
 export AGENT_PATH=`pwd`
 
+# I think we're skipping koa because a new version was breaking tests
```
Comment (michaelgoin) [inline bin/run-versioned-tests.sh:24]: @koa/router 8.03+ breaks segment naming for nested routes.

We need to prob adjust the versions in koa module and publish to limit until fixed (which unfortunately is not straightforward). We might update the comment to that to help future us. Can be a follow-up.

File: bin/ssl.sh (hunk 1 above)
Comment (michaelgoin) [reply bin/ssl.sh:11]: good catch. bash... is not my favorite haha. *watches dev cred decrement by 5*

File: .github/workflows/ci-workflow.yml [hunk 5]
```diff
@@ -75,40 +75,15 @@ jobs:
...
         LASP_SECURE_LICENSE: ${{ secrets.LASP_SECURE_LICENSE }}
-        BENDER_LICENSE: ${{ secrets.BENDER_LICENSE }}
-        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
 
   versioned:
     runs-on: ubuntu-latest
 
-    if: github.repository == 'newrelic/node-newrelic'
-
```
Comment (astormnewrelic) [inline .github/workflows/ci-workflow.yml:111]: what's the intent behind removing the `if: github.repository == 'newrelic/node-newrelic'` conditional?

File: .github/workflows/ci-workflow.yml [hunk 6]
```diff
@@ -75,40 +75,15 @@ jobs:
...
-    - name: Run Versioned Tests
-      run: npm run secrets
       env:
+        TEST_LICENSE: ${{ secrets.TEST_LICENSE }}
+        COLLECTOR_LICENSE: ${{ secrets.COLLECTOR_LICENSE }}
         LASP_LICENSE: ${{ secrets.LASP_LICENSE }}
         LASP_SECURE_LICENSE: ${{ secrets.LASP_SECURE_LICENSE }}
-        BENDER_LICENSE: ${{ secrets.BENDER_LICENSE }}
-        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (astormnewrelic) [inline .github/workflows/ci-workflow.yml:83]: I see we're removing the AWS_ACCESS_KEY_ID variables, which (based on other changes) means we won't run the AWS SDK versioned tests. Presuming this is intentional -- if not see-something/say-something.

File: bin/ssl.sh (hunk 1 above)
Comment (nijotz) [reply bin/ssl.sh:11]: Oh this was def a TIL for me

File: .github/workflows/ci-workflow.yml (hunk 5 above)
Comment (michaelgoin) [reply .github/workflows/ci-workflow.yml:111]: it was only necessary for skipping which this isn't doing anymore. also, it never worked for forked-repo PR's

File: .github/workflows/ci-workflow.yml (hunk 6 above)
Comment (michaelgoin) [reply .github/workflows/ci-workflow.yml:83]: this is only impacting integration tests. the versioned tests still have the secrets.

**Pull request merged**